src/mode/modeHandler.ts

```diff
@@ -486,7 +486,13 @@ export class ModeHandler implements vscode.Disposable {
       var newPosition = new Position(selection.active.line, selection.active.character);
 
       if (newPosition.character >= newPosition.getLineEnd().character) {
-        newPosition = new Position(newPosition.line, Math.max(newPosition.getLineEnd().character - 1, 0));
+        if (this._vimState.currentMode === ModeName.Insert) {
+          // This lets you use arrow keys to move to the end of the line
+          newPosition = new Position(newPosition.line, Math.max(newPosition.getLineEnd().character, 0));
```

Comment:
Is this necessary? The position will already be bounded between 0 and the length of the line anyways.